velox/exec/Unnest.cpp

```diff
@@ -192,13 +192,29 @@ RowVectorPtr Unnest::getOutput() {
       // Construct unnest column using Array elements wrapped using above
       // created dictionary.
       auto unnestBaseArray = currentDecoded.base()->as<ArrayVector>();
-      outputs[outputsIndex++] = identityMapping
-          ? unnestBaseArray->elements()
-          : wrapChild(
-                numElements,
-                elementIndices,
-                unnestBaseArray->elements(),
-                nulls);
+      auto elements = unnestBaseArray->elements();
+
+      if (elements->typeKind() == TypeKind::ROW) {
+        auto size = elements->type()->size();
+        auto rowVector = isDictionary(elements->encoding())
```

Comment:
You need to use DecodedVector here.

Reply:
@pramodsatya : I think Masha is suggesting to use DecodedVector to decode the contents of elements array without using isDictionary(...) on line 199.

Something like
```
DecodedVector rowDecoded(elements);
auto rowVector = rowDecoded->base()->as<RowVector>();
```

Can you try that ?

Reply:
Thanks for the clarification @aditi-pandit, updated.